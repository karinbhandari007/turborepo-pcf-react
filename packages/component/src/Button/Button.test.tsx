import React from 'react';
import { render } from '@testing-library/react';
import Button from './Button';

test('renders Button component correctly', () => {
  const { asFragment } = render(
    <Button
      primary={true}
      size="large"
      backgroundColor="blue"
      label="Click Me"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders secondary Button component correctly', () => {
  const { asFragment } = render(
    <Button
      primary={false}
      size="medium"
      backgroundColor="gray"
      label="Secondary Action"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders small Button component correctly', () => {
  const { asFragment } = render(
    <Button
      primary={false}
      size="small"
      backgroundColor="green"
      label="Small Button"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
