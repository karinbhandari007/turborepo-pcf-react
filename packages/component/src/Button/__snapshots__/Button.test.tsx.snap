// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component correctly 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--large storybook-button--primary"
    style="background-color: blue;"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`renders secondary Button component correctly 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--medium storybook-button--secondary"
    style="background-color: gray;"
    type="button"
  >
    Secondary Action
  </button>
</DocumentFragment>
`;

exports[`renders small Button component correctly 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--small storybook-button--secondary"
    style="background-color: green;"
    type="button"
  >
    Small Button
  </button>
</DocumentFragment>
`;
